adadamp/_dist.py

```diff
@@ -311,3 +312,46 @@ def _get_gradients(
         grads = [grad_fn(idx=idx) for idx in worker_idxs]
         out = dask.compute(*grads)
         return out
+
+class DaskBaseDamper2(DaskBaseDamper):
+    
+    def fit(self, data, target):
```

Comment:
Last time we met, you said you'd like the stats to be saved a class level (rather than outside of the class). The stats would then be received automatically when calling .get_stats.

I am confused about how to implement this on an epoch level though. We said we only wanted stats to be recorded once an epoch, but is that something that is tracked outside of the DaskBaseDamper class? Ie isnt the class itself oblivious to the current epoch? 

The way it is implemented now, I think it will save stats for each batch.

Reply:
Nevermind that is at the epoch levle